src/sentry/monkey/pickle.py

```diff
@@ -117,11 +117,13 @@ def patch_pickle_loaders():
 
     class CompatPickler:
         def __init__(self, *args, **kwargs):
+            # If we don't explicitly pass in a protocol, use DEFAULT_PROTOCOL
             # Enforce protocol kwarg as DEFAULT_PROTOCOL. See the comment above
             # DEFAULT_PROTOCOL above to understand why we must pass the kwarg due
             # to _pickle.
             if len(args) == 1:
-                kwargs["protocol"] = pickle.DEFAULT_PROTOCOL
+                if not kwargs.get("protocol"):
+                    kwargs["protocol"] = pickle.DEFAULT_PROTOCOL
```

Comment:
I think that `kwargs.setdefault("protocol", pickle.DEFAULT_PROTOCOL)` will work here. Only difference is that it won't override a falsey protocol if passed, but I don't think we'd pass one?

Reply:
Wouldn't that also trigger a failure from pickle as well though?